plugins/inputs/opensearch_query/opensearch_query_test.go

```diff
@@ -0,0 +1,774 @@
+			MeasurementName: "measurement16",
+			MetricFields:    []string{"size"},
+			FilterQuery:     "downloads",
+			MetricFunction:  "percentiles",
+			DateField:       "@timestamp",
+			QueryPeriod:     queryPeriod,
+			Tags:            []string{"URI.keyword"},
+			mapMetricFields: map[string]string{"size": "long"},
+		},
+		expectedMetrics: []telegraf.Metric{
+			testutil.MustMetric(
+				"measurement16",
+				map[string]string{"URI_keyword": "/downloads/product_1"},
+				map[string]interface{}{
+					"size_percentiles_values_1.0":  float64(0),
+					"size_percentiles_values_5.0":  float64(0),
+					"size_percentiles_values_25.0": float64(0),
+					"size_percentiles_values_50.0": float64(324),
+					"size_percentiles_values_75.0": float64(337),
+					"size_percentiles_values_95.0": float64(341),
+					"size_percentiles_values_99.0": float64(471.28000000000065),
+					"doc_count":                    int64(263)},
+				time.Date(2018, 6, 14, 5, 51, 53, 266176036, time.UTC),
+			),
+			testutil.MustMetric(
+				"measurement16",
+				map[string]string{"URI_keyword": "/downloads/product_2"},
+				map[string]interface{}{
+					"size_percentiles_values_1.0":  float64(0),
+					"size_percentiles_values_5.0":  float64(0),
+					"size_percentiles_values_25.0": float64(0),
+					"size_percentiles_values_50.0": float64(324),
+					"size_percentiles_values_75.0": float64(339),
+					"size_percentiles_values_95.0": float64(490),
+					"size_percentiles_values_99.0": float64(2677.419999999997),
+					"doc_count":                    int64(237)},
+				time.Date(2018, 6, 14, 5, 51, 53, 266176036, time.UTC),
+			),
+		},
+	},
+}
+
+func setupIntegrationTest(t *testing.T) (*testutil.Container, error) {
+	var err error
+
+	type nginxlog struct {
+		IPaddress    string    `json:"IP"`
+		Timestamp    time.Time `json:"@timestamp"`
+		Method       string    `json:"method"`
+		URI          string    `json:"URI"`
+		Httpversion  string    `json:"http_version"`
+		Response     string    `json:"response"`
+		Size         float64   `json:"size"`
+		ResponseTime float64   `json:"response_time"`
+	}
+
+	container := testutil.Container{
+		Image:        "opensearchproject/opensearch:2.4.1",
+		ExposedPorts: []string{servicePort},
+		Env: map[string]string{
+			"discovery.type": "single-node",
+		},
+		WaitingFor: wait.ForAll(
+			wait.ForLog(".opendistro_security is used as internal security index."),
+			wait.ForListeningPort(nat.Port(servicePort)),
+		),
+	}
+	err = container.Start()
+	require.NoError(t, err, "failed to start container")
+
+	url := fmt.Sprintf("https://%s:%s", container.Address, container.Ports[servicePort])
+
+	o := &OpensearchQuery{
+		URLs:         []string{url},
+		Timeout:      config.Duration(time.Second * 30),
+		Log:          testutil.Logger{},
+		Username:     "admin",
+		Password:     "admin",
+		ClientConfig: tls.ClientConfig{InsecureSkipVerify: true},
+	}
+
+	err = o.connectToOpensearch()
+	if err != nil {
+		return &container, err
+	}
```

Comment:
What is this used for?

Reply:
This creates a client instance to populate the test OpenSearch server with test data.

Reply:
Hmmm you are doing the same `connectToOpensearch` in the tests IIRC. Is it necessary to do it twice?

Reply:
It is not.  Rewrote it to only generate a new `*OpensearchQuery` as part of the test setup